mission/mission_handler.hpp

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ * Copyright (c) 2009-2016, MAV'RIC Development Team
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+
+/*******************************************************************************
+ * \file mission_handler.hpp
+ *
+ * \author MAV'RIC Team
+ * \author Matthew Douglas
+ *
+ * \brief The MAVLink mission planner handler
+ *
+ ******************************************************************************/
+
+
+#ifndef MISSION_HANDLER__
+#define MISSION_HANDLER__
+
+#include "mission/mission_planner.hpp"
+#include "mission/waypoint.hpp"
+
+/*
+ * This mission handler interface. Outine functions that child classes must
+ * implement in order to have the drone achieve the desired mission command.
+ * Child classes will probably pass in a reference to some controller and
+ * pass the desired control commands. If a specific handler should have more
+ * than one possible method of controlling it (e.g. position control, velocity
+ * control), then the mission handler child class can be a template that allows
+ * a template to be inputted instead of a specific control type.
+ */
+
+class Mission_handler
+{
```

Comment:
The argument `Mission_planner& mission_planner` is never used in the following methods:
```cpp
virtual bool can_handle(const Waypoint& wpt) const = 0;
```
```cpp
virtual bool setup(Mission_planner& mission_planner, const Waypoint& wpt) = 0;`
```
```cpp
virtual int update(Mission_planner& mission_planner) = 0;
```
Let's remove it